### DISCUSSION TITLE
Add scrollbar flex item

### DISCUSSION THREAD

**Discussion by wguerram** [tailwindlabs/tailwindcss#9212] [Help] [answered]
Hello, how do I add scrollbar to the flex item, the first column in this case.

Here's the playground
https://play.tailwindcss.com/Buvtpk9an2

I want the scroll bar to appear on the first column not in the whole page.

**Comment by wongjn** [comment, marked as answer]
We will need to make this kind of layout:
```
┌─────────────────────────────┐
│ Header                      │ 
├──────┬──────────────────────┤
│ Side │ Content              │
│ bar  │                      │
│      │                      │
│      │                      │
│      │                      │
└──────┴──────────────────────┘
```
However, we need to make sure the height of this entire thing never exceeds the viewport height, to fulfil your requirement that there must not be a scrollbar for the whole page.

Since you mention flex, we can split it into two flex layouts, one vertical and one horizontal. This is to ensure height for the height adapts to whatever content we throw in there but also the content below does not exceed the bottom of the browser viewport:
```
┌───────────────────────────────┐↑
│ Header                        │ 
├───────────────────────────────┤
│←                             →│
│┌──────┬──────────────────────┐│
││ Side │ Content              ││
││ bar  │                      ││
││      │                      ││
││      │                      ││
││      │                      ││
││      │                      ││
│└──────┴──────────────────────┘│
└───────────────────────────────┘↓
```

We need to add `h-full` to both `<html>` and `<body>` so can we get the viewport height for our top-level vertical flex item (i.e. the `<main>` in your Tailwind Play). We *could* use `min-h-screen` but `100vh` can be a bit finicky – especially with scrollbars or mobile browsers where a toolbar can appear and disappear – so I normally favour inheriting down the `h-full` from `<html>` and `<body>`.

We then add `flex flex-col` to initiate a vertical flex layout on this top-level element, which will contain the header and a wrapper for the sidebar and content.

For this wrapper, we add `flex-1` so that it can grow as much as it likes to fill the `h-full` parent. However [flex items **will not shrink lower than their minimum content size** in the flex direction](https://developer.mozilla.org/en-US/docs/Web/CSS/flex#description), so if one of the children has content that is naturally taller than the space, it will enlarge the space and we will have a layout taller than the `h-full` and the viewport, which would introduce a page scrollbar and not fulfil your requirement. Fortunately, we can get around this by setting `min-h-0` on this wrapper to overcome this.

Next, on this same wrapper for the sidebar and content, we can add `flex` to initiate the horizontal flex layout. The default `align-items` value is `stretch` which is fine for what we want and this will mean both the sidebar and content elements will fill the height of the wrapper, or in other words, the bottom section. We can then add `overflow-y-auto` to **both** of these elements, to ensure that they don't overflow the viewport and thus don't create a page scrollbar.

[Here is a Tailwind Play of your example with the above modifications](https://play.tailwindcss.com/C5VP9AKXDe).
Note: since we don't have access to the `<html>` or `<body>` elements, I added the `h-full` in the CSS file instead.

**Reply by wguerram** [reply]
@wongjn thanks for taking the time to do this great explanation.

Works great, now I have a different issue with a collapse component I found. Since I have the list inside it, when I extend it the scroll bars does not appear. I'll look for another component for that. Thanks for your help.
